### PR TITLE
maintain original order of rows when a row other than the top selecte…

### PR DESCRIPTION
**Pull request by nickpell**
…d row is dragged

Previously, if you selected rows A, B, C and dragged B, the new order would be B, A, C.  This PR fixes that problem.

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js [hunk 1]
```diff
@@ -295,31 +295,37 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
...
-                            return 0;
-                        }
+                    var previousRow = ui.item.prev()[0],
+                        previousIndex = null;
+                    if(previousRow) {
+                        previousIndex = parseInt(previousRow.attributes['data-order'].value);
+                    }
+
+                    var movedIndices = [];
+                    $('.selected-for-sort').map(function (index, element) {
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js]: why are you using `.map` here? I think `.each` might be better for what you are doing.

File: corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js [hunk 2]
```diff
@@ -295,31 +295,37 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
...
-                        i = parseInt(element.children[cur].attributes['data-order'].value);
-                        newList.push(list()[i]);
-                    }
-                    list.splice(0, list().length);
-                    for (i = 0; i < newList.length; i++) {
-                        list.push(newList[i]);
+                    var originalList = list.splice(0, list().length);
+
+                    var insertDraggedElements = function() {
+                        movedIndices.map(function (movedIndex) {
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js]: same comment as above

File: corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js [hunk 3]
```diff
@@ -295,31 +295,37 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
...
+                            list.push(originalList[movedIndex]);
+                        });
+                    };
+
+                    if (previousIndex === null) {
+                        insertDraggedElements();
                     }
+                    originalList.map(function(originalListElement, originalListIndex) {
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js]: "

**Comment by nickpell** [comment]
@proteusvacuum updated

**Review by proteusvacuum — APPROVED** [review]

**Comment by nickpell** [comment]
merging because no test coverage

**Pull request merged**